### ISSUE TITLE
[REVIEW]: pygamma-agreement: Gamma measure for inter/intra-annotator agreement in Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2989]
**Submitting author:** @hadware (<a href="http://orcid.org/0000-0002-8511-1644">Hadrien Titeux</a>)
**Repository:** <a href="https://github.com/bootphon/pygamma-agreement/" target ="_blank">https://github.com/bootphon/pygamma-agreement/</a>
**Version:** v0.1.6
**Editor:** @arfon
**Reviewer:** @faroit, @apiad
**Archive:** <a href="https://doi.org/10.5281/zenodo.4917244" target="_blank">10.5281/zenodo.4917244</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493/status.svg)](https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493"><img src="https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493/status.svg)](https://joss.theoj.org/papers/d54271e471b25775e95ebcfc9bcf2493)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/bootphon/pygamma-agreement/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@faroit & @apiad, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @arfon know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bootphon/pygamma-agreement/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@hadware) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bootphon/pygamma-agreement/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@hadware) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @faroit it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/COLI_a_00227 is OK
- 10.1162/COLI_a_00296 is OK
- 10.1177/014272370002006006 is OK
- 10.1145/2833157.2833162 is OK
- 10.1109/ICASSP40776.2020.9052974 is OK
- 10.1038/s41586-020-2649-2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@faroit - This is the review thread for the paper. All of our communications will happen here from now on.

Please read the "Reviewer instructions & questions" in the first comment above.

Both reviewers have checklists at the top of this thread (in that first comment) with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention https://github.com/openjournals/joss-reviews/issues/2989 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for the review process to be completed within about 4-6 weeks but please make a start well ahead of this as JOSS reviews are by their nature iterative and any early feedback you may be able to provide to the author will be very helpful in meeting this schedule.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon re-invite @apiad as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@apiad please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
@apiad - thanks again for agreeing to review this submission! Please review the guidelines linked above and work though the checklist.

Feel free to raise any issues along the way here.

**Comment by whedon** [comment]
:wave: @faroit, please update us on how your review is going.

**Comment by apiad** [comment]
@arfon I wanted to ask about the depth expected in the technical description of the software. This is the first paper I've reviewed for JOSS and from what I see, papers are usually not expected to go too deeply into the technical or mathematical details of whatever algorithms are implemented, at least not to the same extent as research papers. Am I correct?

**Comment by arfon** [comment]
@apiad - yes that's correct. We need reviewers to verify the functionality of the software and that can happen in a number of ways including some combination of:

- Inspecting and running automated tests.
- Executing test procedures based on test inputs/expected outputs.
- Reviewing the code.

A _deep_ algorithmic review is not generally required but we do want you to be familiar with what the code is doing. Does that sound OK?

**Comment by Rachine** [comment]
Hello, I just noticed these last exchanges about the algorithms we used in our paper. 
We have slides trying to explain with simple vizualisations what the core algorithm is doing for the gamma agreement.  If it is ok for the review process at JOSS @arfon, happy to share them with you @apiad.

**Comment by apiad** [comment]
> @apiad - yes that's correct. We need reviewers to verify the functionality of the software and that can happen in a number of ways including some combination of:
> 
> * Inspecting and running automated tests.
> * Executing test procedures based on test inputs/expected outputs.
> * Reviewing the code.
> 
> A _deep_ algorithmic review is not generally required but we do want you to be familiar with what the code is doing. Does that sound OK?

Thanks! This is very helpful.

**Comment by apiad** [comment]
> Hello, I just noticed these last exchanges about the algorithms we used in our paper.
> We have slides trying to explain with simple vizualisations what the core algorithm is doing for the gamma agreement. If it is ok for the review process at JOSS @arfon, happy to share them with you @apiad.

Yes, that would be very helpful! I 

have already confirmed the functionality claims (by running all tests offline), I'm in the process of running a few manual examples to see the library in action and also check out the performance a little bit. A brief overview of the algorithm's background would be very nice!

**Comment by Rachine** [comment]
> > Hello, I just noticed these last exchanges about the algorithms we used in our paper.
> > We have slides trying to explain with simple vizualisations what the core algorithm is doing for the gamma agreement. If it is ok for the review process at JOSS @arfon, happy to share them with you @apiad.
> 
> Yes, that would be very helpful!

Here is the link to the slides: https://docs.google.com/presentation/d/1dIyx_Xs8oN-9xKE4pQin1EdMN7dD90ga1LaSluZ_mps/edit?usp=sharing
If we need to clarify something, or if you need anything do not hesitate to ping us.

> I have already confirmed the functionality claims (by running all tests offline), I'm in the process of running a few manual examples to see the library in action and also check out the performance a little bit. A brief overview of the algorithm's background would be very nice!

Best,

**Comment by apiad** [comment]
Thanks, @Rachine, this was very enlightening. 

Outside the scope of the submission, I have a question of my own: is there a rule-of-thumb dataset-independent scale for evaluating how good the agreement is? 

E.g., in Kappa, I believe if you are somewhere above 0.7 it is considered a "good enough" agreement. Of course these values are always going to be very subjective, but in the end, measuring what is "good enough" is always gonna rest on some degree of subjectivity. 

I guess my questions go along the line of is it possible to give the user a qualitative measure of the quality of the agreement, even if it's highly subjective or is this totally dependent on each dataset, and in that case, are there rules-of-thumb that can be used to determine these levels of quality for a specific dataset?

**Comment by Rachine** [comment]
> Thanks, @Rachine, this was very enlightening.
> 
> Outside the scope of the submission, I have a question of my own: is there a rule-of-thumb dataset-independent scale for evaluating how good the agreement is?

This is a very interesting and important question! We have been asked this question by linguists and speech pathologists also.  
The full answer is that we know that only by experience, as the kappa, and this has not been properly measured with psychophysics.  We are launching a collaboration with psychophysics experts, with more experience in perceptual differences to obtain this scale for the Gamma. 

For example, we found that to get 0.6-0.8 in practice is really good for real annotations of speech corpora of turn-takings. 

Above 0.8 was never found so far for speech annotations, such as type of vocalization; speaker turns. 

For instance, we found out that Child-directed speech in Non-native language was harder to annotate. 

Maybe, we should gather a table with all Gamma measurements that have been done so far, so that can people can contribute or be able to compare.

> 
> E.g., in Kappa, I believe if you are somewhere above 0.7 it is considered a "good enough" agreement. Of course these values are always going to be very subjective, but in the end, measuring what is "good enough" is always gonna rest on some degree of subjectivity.
> 
> I guess my questions go along the line of is it possible to give the user a qualitative measure of the quality of the agreement, even if it's highly subjective or is this totally dependent on each dataset, and in that case, are there rules-of-thumb that can be used to determine these levels of quality for a specific dataset?

**Comment by apiad** [comment]
@Rachine thanks again for the explanation! I think it would be a very good idea to make these insights public to your library's users, dunno if with some comments in the docs or a repository where you collect these measurements, as you suggested.

**Comment by apiad** [comment]
@arfon taking into consideration your comments, my exchanges with the author, and having checked the functional claims and read the submission paper and the library source code (which is conveniently commented and suitably organized), I consider this submission meets the quality standards for publishing in JOSS.

I sincerely appreciate the opportunity to collaborate, and please don't hesitate to contact me again in the future if you consider it helpful. 

Big congrats to the authors for the great work done with this library.

**Comment by hadware** [comment]
Hey @faroit, I see that some of your review points for our library still remain unsolved. Do you need more information from us to be able to resolve them? Do you see any improvement to our work that you think are essential for it to be on par with JOSS's standards?

**Comment by faroit** [comment]
@hadware I checked the last remaining issues for this review. I think this is looking good. Can you please address the minor issues I raised in https://github.com/bootphon/pygamma-agreement/issues/24 before we can proceed?

**Comment by arfon** [comment]
@hadware – At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/COLI_a_00227 is OK
- 10.1162/COLI_a_00296 is OK
- 10.1177/014272370002006006 is OK
- 10.1145/2833157.2833162 is OK
- 10.1109/ICASSP40776.2020.9052974 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.18653/v1/D19-1565 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by hadware** [comment]
Ok @arfon , will do! Should I just copy the DOI here, or is there a specific way to do this with @whedon ?

**Comment by arfon** [comment]
Just copy here. Thanks @hadware.

**Comment by hadware** [comment]
All right: 

10.5281/zenodo.4917244

Here's the ZENODO DOI, and here's the package's page : https://zenodo.org/record/4917244 .
I hope this fits the bill.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.4917244 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4917244" target="_blank">10.5281/zenodo.4917244</a> is the archive.

**Comment by arfon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/COLI_a_00227 is OK
- 10.1162/COLI_a_00296 is OK
- 10.1177/014272370002006006 is OK
- 10.1145/2833157.2833162 is OK
- 10.1109/ICASSP40776.2020.9052974 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.18653/v1/D19-1565 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2380

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2380, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi @hadware! Can I verify the version you want associated with this submission before we wrap up?

**Comment by kthyng** [comment]
@hadware a few comments on your paper:

* is this how you usually display your affiliations?
* first sentence under Statement of Need: the reference shouldn't be in parentheses — remove the [] around the reference.
* Same with the final sentence in that paragraph and several more times in the paper. Please update all of these.

**Comment by hadware** [comment]
Hey @kthyng ! 

The current version (0.1.6) is good :)

Sorry for the refs, i thought the brackets were mandatory to render a citation. I'll update those and tell you when it's done.

**Comment by hadware** [comment]
Sorry again for the double message: yes, this is how my affiliation is usually displayed.

**Comment by kthyng** [comment]
@whedon set v0.1.6 as version

**Comment by whedon** [comment]
OK. v0.1.6 is the version.

**Comment by kthyng** [comment]
ok let me know when you're done with paper changes.

**Comment by hadware** [comment]
All right @kthyng : refs are fixed (it does look much cleaner now). @arfon I merged your PR (thanks again for this, super helpful!). I did leave a small question on the PR, you might want to check that out.

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02989/joss.02989/10.21105.joss.02989.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by arfon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/COLI_a_00227 is OK
- 10.1162/COLI_a_00296 is OK
- 10.1177/014272370002006006 is OK
- 10.1145/2833157.2833162 is OK
- 10.1109/ICASSP40776.2020.9052974 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.18653/v1/D19-1565 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2386

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2386, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1403680812181168130) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2387
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02989](https://doi.org/10.21105/joss.02989)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@faroit, @apiad – many thanks for your reviews here! JOSS relies upon the volunteer efforts of folks like you and we simply wouldn't be able to do this without you ✨ 

@hadware – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02989/status.svg)](https://doi.org/10.21105/joss.02989)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02989">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02989/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02989/status.svg
   :target: https://doi.org/10.21105/joss.02989
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02989/status.svg)](https://doi.org/10.21105/joss.02989)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by hadware** [comment]
Thanks to all for your help and feedback with this paper :) I've already volunteered to be a reviewer to JOSS, i'll be waiting for new packages to review :)

**Comment by apiad** [comment]
> @faroit, @apiad – many thanks for your reviews here! JOSS relies upon the volunteer efforts of folks like you and we simply wouldn't be able to do this without you 
> 
> @hadware – your paper is now accepted and published in JOSS

It has been my pleasure! Count me on whenever you need, I'll do my best to be available :smiley:
